harness_tests/diagnostic_test/main1.c

```diff
@@ -1,205 +1,478 @@
 /**
- * Diagnostic test to verify hardware functionality for EPS. This consists of: 
- *      - taking measurements with devices and making sure they are within 
- *          expected range
- *      - turning on a heater shuld increase current from boost and battery
- *      - imu gyro vals should be small numbers
- */
+ * Diagnostic test to verify hardware functionality for EPS.
+ *      - See Bus Architecture Specification (section 5- diagnostic tests) for more information
+ **/
+
 #include <stdbool.h>
 
 #include <adc/adc.h>
 #include <conversions/conversions.h>
 #include <uart/uart.h>
 #include <test/test.h>
+#include <can/data_protocol.h>
 
 #include "../../src/devices.h"
 #include "../../src/heaters.h"
 #include "../../src/imu.h"
 #include "../../src/measurements.h"
 #include "../../src/shunts.h"
-
-
-double read_voltage(uint8_t channel) {
-    fetch_adc_channel(&adc, channel);
-    uint16_t raw_data = read_adc_channel(&adc, channel);
-    double voltage = adc_raw_to_circ_vol(raw_data, ADC_VOL_SENSE_LOW_RES, ADC_VOL_SENSE_HIGH_RES);
-    return voltage;
+#include "../../src/can_commands.h"
+
+uint8_t tx_msg[8] = {0x00};
+
+/* Helper function for generating and appropriately dequeueing rx_housekeeping message */
+void construct_rx_msg_hk(uint8_t field_num){
+    uint8_t rx_msg[8] = {0x00};
+    rx_msg[2] = CAN_EPS_HK;
+    rx_msg[3] = field_num;
+    enqueue(&can_rx_msg_queue, rx_msg);
```

Comment:
Assert here that there is 1 message in the RX queue and 0 in the TX queue (for sanity)